packages/editor/src/store/actions.js

```diff
@@ -396,3 +722,12 @@ export const exitFormattedText = getBlockEditorAction( 'exitFormattedText' );
 export const insertDefaultBlock = getBlockEditorAction( 'insertDefaultBlock' );
 export const updateBlockListSettings = getBlockEditorAction( 'updateBlockListSettings' );
 export const updateEditorSettings = getBlockEditorAction( 'updateEditorSettings' );
+
+// default export of generator actions.
```

Comment:
I understand how you're using this for the unit tests, but it does not seem sensible to me that a default export from this file be a subset of generator actions.

Would it be possible instead to consider for the purposes of `jest.spyOn`, to use the `import * as actions from '../actions';` syntax?

Reply:
doh, ya, I'll see if that works sufficiently for tests.

Reply:
Turns out doing `import * as actions` and then `jest.spyOn( actions, 'savePost')` does not work.

Reply:
> Turns out doing `import * as actions` and then `jest.spyOn( actions, 'savePost')` does not work.

Why not?

Reply:
No idea.  I tried it:

```js
import * as actions from '../actions';

// .... later in the test
describe( 'autosave()', () => {
		let savePostSpy;
		beforeAll( () => savePostSpy = jest.spyOn( actions, 'savePost' ) );
		afterAll( () => savePostSpy.mockRestore() );
		// autosave is mostly covered by `savePost` tests so just test the correct call
		it( 'calls savePost with the correct arguments', () => {
			const fulfillment = actions.autosave();
			fulfillment.next();
			expect( savePostSpy ).toHaveBeenCalled();
			expect( savePostSpy ).toHaveBeenCalledWith( { isAutosave: true } );
		} );
	} );
```

The test fails with the mock not getting called.  When I use `generatorActions` the mock works as expected.  I don't doubt there may be something I'm doing wrong but I'm guessing its because the mock isn't actually getting invoked by the autosave call.

Reply:
I sense it should be possible. The issue is likely that the spy creates a new reference which is different than the one called internal to the underlying `actions.js` file.

But I also wonder: Maybe rather than testing that it was called, we could test that the result of calling autosave would be the same as the result of calling `savePost` with the expected arguments?

Something like:

```js
it( 'calls savePost with the correct arguments', () => {
	expect( Array.from( autosave() ) ).toEqual( Array.from( savePost( { isAutosave: true } ) ) );
} );
```

Reply:
Not sure how practical/useful that will be because they are both generators.

Reply:
So one thing I _could_ do here is instead of yielding the `savePost` generator from `autosave`, I can just yield the `dispatch` control.  That way it's more easily testable and I can get rid of the `generatorActions` default export.